component/commons/handler/src/main/java/org/wso2/carbon/identity/password/expiry/validation/handler/PasswordExpiryValidationHandler.java

```diff
@@ -0,0 +1,146 @@
+package org.wso2.carbon.identity.password.expiry.validation.handler;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * Handler to check whether the password is expired. This will throw an identity event exception if password is expired.
+ */
+public class PasswordExpiryValidationHandler extends AbstractEventHandler {
+    private static final Log log = LogFactory.getLog(PasswordExpiryValidationHandler.class);
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        String username = (String) event.getEventProperties().get(IdentityEventConstants.EventProperty.USER_NAME);
+        String tenantDomain = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.TENANT_DOMAIN);
+        UserStoreManager userStoreManager = (UserStoreManager) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+
+        try {
+            if (isPasswordExpired(tenantDomain, tenantAwareUsername, userStoreManager)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("User: " + username + " password is expired.");
+                }
+                throw new IdentityEventException("Password expired for user with username: " + tenantAwareUsername);
+            }
+        } catch (UserStoreException e) {
+            throw new IdentityEventException("UserStore Exception occurred", e);
```

Comment:
The exception itself is a userstore exception. So it's unnecessary to say a userstore exception occurred. Better to give some more context data instead.